### PR TITLE
Flow Chapter

### PR DESCRIPTION
**Pull request by eric-czech**
Hey @hammer ,

This PR has a consolidated version of the flow chapter and there are still a bunch of little things to finish, but this is at least the large majority of what I was going to try to put in there. 

It's slow to build (~13 mins total) but I think I can get that down a good bit.  Also, I was exporting paperpile citations into a separate ```flow.bib``` file but let me know if you'd rather work from the same paperpile folder or something like that so there's only one citation db.

I addressed a few of your notes in https://github.com/hammerlab/t-cell-data/issues/5 but I'll keep chipping away at the rest too.

### CODE REVIEW

**Comment by hammer** [comment]
Cool looks like the PR builds on Travis! Need to figure out how to deploy to a staging site for a branch though...

File: Flow.Rmd [hunk 1]
```diff
@@ -123,13 +169,407 @@ gating(gt, gs)
...
+
+```{r flow-OMIP-030-figure, results='asis', fig.align='center', fig.cap='Gating strategy and cell differentiation overlay from OMIP-030', echo=FALSE}
+knitr::include_graphics('images/flow-OMIP030-model.png')
+```
+
+### Extracting Single Cell Data
+
+Identify terminal nodes (i.e. cell populations) in the workflow which were, by convention, prefixed with either ```CD4+``` or ```CD8+``` and then use those nodes to extract single-cell data into a data frame (useful for custom visualizations).
+
+```{r flow-cell-metadata, results='asis'}
```
Comment (hammer) [inline Flow.Rmd]: The results printed by this chunk are pretty ugly and my brain parsed them as debug spew. Maybe format a little? One idea would be a 2x2 table with cell types, [[list of cell types]] as the top row and cell states, [[list of cell states]] as the bottom row. I'm open to anything that looks a little better.

File: Flow.Rmd [hunk 2]
```diff
@@ -123,13 +169,407 @@ gating(gt, gs)
...
-```{r plot-cell-types, fig.height=8, fig.width=8}
+```{r flow-plot-cell-types, fig.height=8, fig.width=8}
 autoplot(gs[[1]], strip.text = "gate", bins=100, merge=F, axis_inverse_trans=F) + flow_theme
 ```
 
 This should then be comparable to what was in the [OMIP-021 publication](https://onlinelibrary.wiley.com/doi/full/10.1002/cyto.a.22475), and here are relevant figures demonstrating the similarity of the cell subsets captured:
 
-```{r OMIP-021-figure, fig.align='center', fig.cap='Gating strategy illustrated with annotated figures from OMIP-021', echo=FALSE}
-knitr::include_graphics('images/OMIP021.png')
+```{r flow-OMIP-021-figure, results='asis', fig.align='center', fig.cap='Gating strategy illustrated with annotated figures from OMIP-021', echo=FALSE, out.width = '100%'}
```
Comment (hammer) [inline Flow.Rmd]: Was `results='asis'` the trick to getting this to display correctly?

File: Flow.Rmd [hunk 3]
```diff
@@ -27,26 +36,63 @@ flow_theme <- theme_bw() + theme(
...
+tibble(
+  id=studies %>% map_chr(~ .@id),
+  name=studies %>% map_chr(~ .@name),
+  panel=name %>% str_extract('(?<=OMIP-)\\d+') %>% as.integer,
+  has_flowjo_ws=studies %>% map_lgl(~ map(.@attachments, ~.@name) %>% str_detect('\\.jo|\\.wsp') %>% any)
+) %>% 
+  filter(has_flowjo_ws) %>% 
+  select(-has_flowjo_ws) %>% 
+  arrange(panel) %>%
+  knitr::kable()
```
Comment (hammer) [inline Flow.Rmd]: I think this table (and others) need a [horizontal scroll box](https://cran.r-project.org/web/packages/kableExtra/vignettes/awesome_table_in_html.html#scroll_box) to see all columns.

**Review by hammer — APPROVED** [review]
Looks good to me, with a few potential styling changes!

File: Flow.Rmd (hunk 2 above)
Comment (eric-czech) [reply Flow.Rmd]: Yep, that seemed to do the trick.

File: Flow.Rmd (hunk 1 above)
Comment (eric-czech) [reply Flow.Rmd]: Agreed that it's gross -- I'll find something prettier

File: Flow.Rmd (hunk 3 above)
Comment (eric-czech) [reply Flow.Rmd]: Ah very cool, did not realize you could do that with knitr.  I'll make sure to add it.

**Pull request merged**